src/Program.cs

```diff
@@ -361,15 +361,7 @@ private static bool TryLoadMSBuild([NotNullWhen(returnValue: true)] out string?
             {
                 // Since we are running as a dotnet tool we should be able to find an instance of
                 // MSBuild in a .NET Core SDK.
-                var msBuildInstance = Build.Locator.MSBuildLocator.QueryVisualStudioInstances().First();
-
-                // Since we do not inherit msbuild.deps.json when referencing the SDK copy
-                // of MSBuild and because the SDK no longer ships with version matched assemblies, we
-                // register an assembly loader that will load assemblies from the msbuild path with
-                // equal or higher version numbers than requested.
-                LooseVersionAssemblyLoader.Register(msBuildInstance.MSBuildPath);
-                Build.Locator.MSBuildLocator.RegisterInstance(msBuildInstance);
-
+                var msBuildInstance = Build.Locator.MSBuildLocator.RegisterDefaults();
```

Comment:
No longer need to load assemblies ourselves and RegisterDefaults does the right thing by returning the SDK pinned in the global.json.